ui/src/app/pages/signed-in/signed-in.tsx

```diff
@@ -0,0 +1,171 @@
+import {environment} from 'environments/environment';
+import {useEffect, useState} from 'react';
+
+const styles = reactStyles({
+  appContainer: {
+    width: '100%',
+    paddingRight: '0.6rem',
+    paddingLeft: '0.6rem',
+    flexGrow: 1,
+    /* Needed for absolute positioned child elements, e.g. spinner. */
+    position: 'relative'
+  },
+  backgroundImage: {
+    backgroundImage: 'url(\'/assets/images/BG-Pattern.png\')',
+    backgroundSize: '80px', /* half the size of the image */
+    width: '100%',
+    height: '100%',
+    zIndex: -1,
+    position: 'absolute'
+  },
+});
+
+const checkOpsBeforeUnload = (e) => {
+  if (Object.keys(compoundRuntimeOpStore.get()).length > 0) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload
+    e.preventDefault();
+    e.returnValue = '';
+  }
+};
+
+interface Props extends WithSpinnerOverlayProps {
+  onSignOut: () => {};
+  signOut: () => {};
+}
+
+export const SignedIn = (props: Props) => {
+  useEffect(() => props.hideSpinner(), []);
+
+  const [hideFooter, setHideFooter] = useState(null);
+  const [cdrVersionsInitialized, setCdrVersionsInitialized] = useState(false);
+  const [serverConfigInitialized, setServerConfigInitialized] = useState(false);
+  const [subscriptions, setSubscriptions] = useState([]);
+
+  const serverConfig = useStore(serverConfigStore);
+  const cdrVersions = useStore(cdrVersionStore);
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', checkOpsBeforeUnload);
+
+    return () => {
+      window.removeEventListener('beforeunload', checkOpsBeforeUnload);
+    };
+  }, []);
+
+  useEffect(() => {
+    const subs = [];
+    // This handles detection of Angular-based routing data.
+    subs.push(routeConfigDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    // This handles detection of React-based routing data. During migrations,
+    // we assume React routing data will be set deeper/later in the component
+    // hierarchy, therefore it will generally take precedence over React.
+    subs.push(routeDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    setSubscriptions(subs);
+
+    return () => {
+      /*
+       * TODO RW-1920: we will only need one subscription after routeConfigDataStore and
+       *  routeDataStore are collapsed, and then we won't have to duck-type this
+       */
+      for (const s of subscriptions) {
+        s.unsubscribe();
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    /*
+     * TODO RW-6726 We want to block app rendering on the presence of server config data
+     *  so that route components don't try to lookup config data before it's available.
+     *  We should holistically figure out a Reacty way to do this.
+     *  See discussion on https://github.com/all-of-us/workbench/pull/4713
+     */
+    const checkStoresLoaded = async() => {
+      if (serverConfig.config) {
```

Comment:
I think there are some issues here, though it may be working based on the order of how things are currently executing at runtime:

1. Since `serverConfig` is state, and `serverConfigInitialized` is derived from it, `serverConfigInitialized` is duplicating that state - it would also be possible to just derive this as the top of the function as a constant for the sake of minimizing state.
2. This useEffect callback references `serverConfig`, but declares props as `[]`, so this only ever runs once at first render. If this works currently, I think that implies that this entire component is only ever rendered after `serverConfigStore` is loaded, so we could either enshrine that (and remove the distinction of `serverConfigLoaded`), or update the useEffect